### PR TITLE
isolate: init at 1.10

### PR DESCRIPTION
**Pull request by virchau13**
###### Description of changes

Adds `isolate`, a sandbox for securely executing untrusted programs (especially in a competitive programming environment.)
https://github.com/ioi/isolate

This package just contains a binary `bin/isolate` and a man page `share/man/man1/isolate.1`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/isolate/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildPhase = ''
+    make isolate isolate.1
+  '';
+
+  installPhase = ''
+    mkdir -p $out/{bin,share/man/man1}
+    cp ./isolate $out/bin/
+    cp ./isolate.1 $out/share/man/man1/
+  '';
```
Comment (AndersonTorres) [inline pkgs/os-specific/linux/isolate/default.nix:27]:
```suggestion
  installPhase = ''
    runHook preInstall

    install -Dm755 ./isolate $out/bin/
    installManPage isolate.1

    runHook postInstall
  '';
```

File: pkgs/os-specific/linux/isolate/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/os-specific/linux/isolate/default.nix:29]:
```suggestion
  meta = {
```

Don't tell anyone, but nested `with` is mad.

File: pkgs/os-specific/linux/isolate/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    description = "Sandbox for securely executing untrusted programs";
+    homepage = "https://github.com/ioi/isolate";
+    license = with licenses; [ gpl2 ];
```
Comment (AndersonTorres) [inline pkgs/os-specific/linux/isolate/default.nix:32]:
```suggestion
    license = lib.licenses.gpl2Plus;
```

File: pkgs/os-specific/linux/isolate/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    maintainers = with maintainers; [ virchau13 ];
```
Comment (AndersonTorres) [inline pkgs/os-specific/linux/isolate/default.nix:33]:
```suggestion
    maintainers = with lib.maintainers; [ virchau13 ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -39436,4 +39436,6 @@ with pkgs;
   gnss-share = callPackage ../servers/gnss-share { };
 
   ali = callPackage ../tools/networking/ali { };
+
+  isolate = callPackage ../os-specific/linux/isolate { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:39440]:
```suggestion
  isolate = callPackage ../tools/misc/isolate { };
```

`os-specific` is for low-level programs that are not made for other environments - things like device drivers, kernel patches...

When in doubt, just throw the things under `tools/misc/`

**Review by AndersonTorres — APPROVED** [review]

**Review by lilyinstarlight — APPROVED** [review]

**Pull request merged**